py/server/deephaven/table.py

```diff
@@ -1034,6 +1077,28 @@ def avg_by(self, by: Union[str, Sequence[str]] = None) -> Table:
         except Exception as e:
             raise DHError(e, "table avg_by operation failed.") from e
 
+    def weighted_avg_by(self, wcol: str, by: Union[str, Sequence[str]] = None) -> Table:
```

Comment:
I think there is a question about how best to make things pythonic in the API.  Here we have `weighted_avg_by` which mirrors the java API and is quite verbose.  I looked into what numpy is doing for weighted averages, and it just adds a named argument (https://numpy.org/doc/stable/reference/generated/numpy.average.html).  Should we consider this path?  Similar question for `abs`.

Reply:
This is not a very clear-cut case. While being concise is definitely a trait of being Pythonic, being explicit is also one. I have considered to use wavg_by, wsum_by, asum_by for the method names but not happy with 'asum_by' and gave up. Another factor influencing the picking of these names is that we already have 'weighted_avg/sum' in agg.py.

Reply:
I'm not sure how `avg_by(wcol="X", by=["A","B"])` is not explicit.  Having said that, I'm not opposed to what is here.  I'm just pushing the question to confirm that what is here is the best way to do the API for a user.

Reply:
I am not saying it is not very explicit but maybe less so than weighted_avg_by(). I think I can live with the names especially since they are sorta consistent with what's in agg already.